CHANGELOG.md

```diff
@@ -2,6 +2,7 @@
 
 ## unreleased
 
+- Clarify Unicode and UTF-8 references.
```

Comment:
I wonder whether this even needs to be mentioned, considering that it doesn't change anything substantially.

Reply:
Since the body of the text has a substantial number of changes, I thought it was worth including. We could drop that line if it's really unnecessary.

Reply:
I could one new sentence (about the "well-formed code-unit sequence"), two deleted words (2x "valid") and the reference to "binary-to-text encoding" (which technically isn't Unicode-related). So, while I wouldn't strongly object, in my opinion this doesn't warrant a changelog entry.

Reply:
@ChristianSi Changelog entries aren't a finite resource.

Reply:
If the new changelog entry conflicted with an existing one, I'd change or remove the new one. But since this is not the case, I'm keeping it.